boa_unicode/src/lib.rs

```diff
@@ -0,0 +1,84 @@
+//! This crate implements the extension to query if a char belongs to a particular unicode identifier property.
+//! Unicode version: 13.0.0
+//!
+//! More information:
+//!  - [Unicode® Standard Annex #31][uax31]
+//!
+//! [uax31]: http://unicode.org/reports/tr31
+
+mod tables;
+
+use unicode_general_category::{get_general_category, GeneralCategory};
+pub trait UnicodeProperties: Sized + Copy {
+    /// Returns `true` if this value is a member of "ID_Start".
+    fn is_id_start(self) -> bool;
+
+    /// Returns `true` if this value is a member of "ID_Continue".
+    fn is_id_continue(self) -> bool;
+
+    /// Returns `true` if this value is a member of "Other_ID_Start".
+    fn is_other_id_start(self) -> bool;
+
+    /// Returns `true` if this value is a member of "Other_ID_Continue".
+    fn is_other_id_continue(self) -> bool;
+
+    /// Returns `true` if this value is a member of "Pattern_Syntax".
+    fn is_pattern_syntax(self) -> bool;
+
+    /// Returns `true` if this value is a member of "Pattern_White_Space".
+    fn is_pattern_whitespace(self) -> bool;
+}
```

Comment:
This trait needs a bit of trait-level documentation. Mostly to explain what kinds of values implement it, why do they implement it and in what contexts is this trait useful.